src/main/java/rx/exceptions/CompositeException.java

```diff
@@ -293,4 +292,25 @@ public String getMessage() {
             }
         }
     }
+
+    /**
+     * Returns the root cause of {@code e}. If {@code e.getCause()} returns {@null}, just return {@code e} itself.
+     *
+     * @param e the {@link Throwable} {@code e}.
+     * @return The root cause of {@code e}. If {@code e.getCause()} returns {@null}, just return {@code e} itself.
+     */
+    private Throwable getRootCause(Throwable e) {
+        Throwable root = e.getCause();
+        if (root == null) {
+            return e;
+        } else {
+            while(true) {
+                if (root.getCause() == null) {
```

Comment:
I have seen cases where a `Throwable` will return `this` for `getCause()` which can cause infinite loops in code like this.

I think you want:
```java
while (true) {
  Throwable cause = root.getCause();
  if (cause == null || cause == root) {
    return root;
  }
  root = cause;
}
```
as a result instead.